zeroize/src/lib.rs

```diff
@@ -858,6 +886,28 @@ mod tests {
         assert!(as_vec.iter().all(|byte| *byte == 0));
     }
 
+    #[cfg(feature = "std")]
+    #[test]
+    fn zeroize_c_string() {
+        let mut cstring = CString::new("Hello, world!").expect("CString::new failed");
+        let clone = cstring.clone();
+        let orig_len = cstring.as_bytes().len();
+        let orig_ptr = cstring.as_bytes().as_ptr();
+        cstring.zeroize();
+        // This doesn't quite test that the original memory has been cleared, but only that
+        // cstring now owns an empty buffer
+        assert!(cstring.into_bytes().is_empty());
+        unsafe {
+            // DEFINITELY UB! We reconstruct the original buffer which should be deallocated after
+            // the .zeroize()
+            let orig_buf = std::slice::from_raw_parts(orig_ptr, orig_len);
```

Comment:
I used `read_volatile` instead of a deref as otherwise only the first byte was checked when testing in debug mode.

When compiling with `--release` apparently the whole loop is optimized away, even with `read_volatile` (I must admit, I'm not a 100% clear on what exactly `read_volatile` ensures).